test/EndToEndTests/Common/Microsoft.OData.Client.E2E.TestCommon/Samples/Tests/SampleTest.cs

```diff
@@ -0,0 +1,45 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="SampleTest.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.OData;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.OData.Client.E2E.TestCommon.Samples.Client;
+using Microsoft.OData.Client.E2E.TestCommon.Samples.Server;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.TestCommon.Samples.Tests
+{
+    public class SampleTest : EndToEndTestBase<SampleTest.TestsStartup>
+    {
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(SampleController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("actionresult", SampleEdmModel.GetEdmModel()));
+            }
+        }
+        public SampleTest(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+        }
+
+        [Fact]
+        public void ExecuteOverloadedOperations()
+        {
+            Uri uri = new Uri(Client.BaseAddress, "actionresult/RetrieveProduct");
+            CreateDataServiceContext = uri => new DefaultContainer(uri);
+            var context = CreateDataServiceContext(uri);

```

Comment:
What's the value of wrapping the constructor in a delegate and createing the context via the delegate instead of calling the constructor directly as follows?:
```suggestion
            var context = new DefaultContainer(uri);

```

Reply:
Yea. There is no need for the delegate here. We can do away with it.